package.json

```diff
@@ -6,15 +6,15 @@
   ],
   "packageManager": "yarn@1.22.19",
   "devDependencies": {
-    "@typescript-eslint/eslint-plugin": "^5.59.1",
-    "@typescript-eslint/parser": "^5.59.1",
-    "eslint": "^8.39.0",
+    "@typescript-eslint/eslint-plugin": "^5.60.1",
+    "@typescript-eslint/parser": "^5.60.1",
+    "eslint": "^8.43.0",
     "eslint-config-prettier": "^8.8.0",
     "prettier": "^2.8.8",
     "tslint": "^6.1.3",
     "tslint-config-prettier": "^1.18.0",
     "tslint-plugin-prettier": "^2.3.0",
-    "typescript": "^5.0.4"
+    "typescript": "5.0.4"
```

Comment:
Did a little bit of dependency bumping since I was adding new packages to avoid pinning the new packages to old versions.

Ran into a couple of small issues that I'll sort out separately:

1. `@js-temporal/polyfill` v0.4.4 now gives slightly different results for one particular range test.
2. `typescript` v5.1.x changed some behavior w.r.t. to `undefined` and `void` which has the side effect shown in #635 

Both are totally fixable, but I didn't want to add noise to this PR.